RecoLocalTracker/SiPixelClusterizer/plugins/gpuCalibPixel.h

```diff
@@ -0,0 +1,67 @@
+#ifndef RecoLocalTracker_SiPixelClusterizer_plugins_gpuCalibPixel_h
+#define RecoLocalTracker_SiPixelClusterizer_plugins_gpuCalibPixel_h
+
+#include <cstdint>
+#include <cstdio>
+
+#include "CondFormats/SiPixelObjects/interface/SiPixelGainForHLTonGPU.h"
+#include "HeterogeneousCore/CUDAUtilities/interface/cuda_assert.h"
+
+#include "gpuClusteringConstants.h"
+
+namespace gpuCalibPixel {
+
+  constexpr uint16_t InvId = 9999;  // must be > MaxNumModules
+
+  constexpr float VCaltoElectronGain = 47;         // L2-4: 47 +- 4.7
```

Comment:
@VinInn, as you maybe know we changed the gain calibration scheme (for Run3 only) in this PR: https://github.com/cms-sw/cmssw/pull/29333 by absorbing the VCal calibration from within the gain calibration DB payloads. 
Since CMSSW_11_1_0_pre7 this code is not compatible anymore with the gain calibration in Global Tag.
This leads to funky pixel cluster charge:

![image](https://user-images.githubusercontent.com/5082376/85121100-7f263a80-b224-11ea-8020-f7de7c41577c.png)

**IF** we have to support Run-2 *and* Run-3 production as well these numbers can't stay hard-coded as here. Please advice.

Reply:
my advice would be to change the payload for run< 3 as well.
How the code for run < 3 looks like? please point me to the calibration code for run <3

Reply:
we wanted to avoid changing also the run < 3 payloads because it's a labour intensive procedure involving tens of tags and scores of GTs...
The numbers you have currently in the patatrack branch are OK for Run1 and Run2, while for Run3 we have just:
```python
VCaltoElectronGain = 1, # all gains=1, pedestals=0
VCaltoElectronGain_L1 = 1,
VCaltoElectronOffset = 0,
VCaltoElectronOffset_L1 = 0
```
all the conversion factors are already included in the Run3 gain DB payloads (all gains are multiplied and shifted according to the slope and offsets which were earlier passed as configuration parameters in the clusterizer).
You might also consult this talk: https://indico.cern.ch/event/879474/contributions/3855389/

@dkotlins @IzaakWN @tsusa @tvami FYI.

Reply:
@VinInn 

>  (will have a look, will go in some sort of config, I suspect)

yes it is here: https://github.com/cms-sw/cmssw/blob/master/RecoLocalTracker/SiPixelClusterizer/python/SiPixelClusterizer_cfi.py#L7-L28

Reply:
I will introduce a flag isRun3 setting it using the modifier and take proper action in gpuCalib

Reply:
https://github.com/cms-patatrack/cmssw/pull/492

Reply:
@mmusich, thanks for the warning.

@VinInn would it make sense to make the four values customisable from python, like in `SiPixelClusterizer_cfi.py ` ?

I don't mind either way, simply trying to think what is more "future proof".

Reply:
@mmusich, is there a difference in data vs MC ?

Reply:
> @mmusich, is there a difference in data vs MC ?

no.

Reply:
Thanks.

Reply:
I do not like to have in config values that shall be coherent (this is production not a testbeam)
I prefer a flag   and values (not in DB), hardcoded.